src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -1357,6 +1373,42 @@ export default class Fms {
     }
 
     /**
+     * Regenerate SID leg based on current runway assignment
+     *
+     * @for Fms
+     * @method _regenerateSidLeg
+     */
+    _regenerateSidLeg() {
+        const sidLegIndex = this._findLegIndexForProcedureType(PROCEDURE_TYPE.SID);
+
+        if (sidLegIndex === INVALID_INDEX) {
+            return;
```

Comment:
It would be a problem, because we're operating under the assumption that it already exists, and then we find it and regenerate it with the new runway.